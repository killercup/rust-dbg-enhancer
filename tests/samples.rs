use insta::assert_debug_snapshot_matches;
use rust_dbg_enhancer::enhance;

#[test]
fn pest_pairs() {
    assert_debug_snapshot_matches!(
        "struct",
        enhance(r#"Pair { rule: xkb_types_item }"#).unwrap()
    );

    assert_debug_snapshot_matches!(
        "array",
        enhance(r#"[Pair { rule: xkb_types_item }]"#).unwrap()
    );

    assert_debug_snapshot_matches!(
        "full",
        enhance(r#"[Pair { rule: xkb_types_item, span: Span { str: "override type \"TWO_LEVEL\" {\n    modifiers = Shift+Lock;\n    map[Shift] = Level2;\n    preserve[Lock]= Lock;\n    level_name[Level1] = \"Base\";\n    level_name[Level2] = \"Shift\";\n};", start: 50, end: 226 }, inner: [Pair { rule: type_item, span: Span { str: "override type \"TWO_LEVEL\" {\n    modifiers = Shift+Lock;\n    map[Shift] = Level2;\n    preserve[Lock]= Lock;\n    level_name[Level1] = \"Base\";\n    level_name[Level2] = \"Shift\";\n};", start: 50, end: 226 }, inner: [Pair { rule: type_item_prefix, span: Span { str: "override", start: 50, end: 58 }, inner: [] }, Pair { rule: string, span: Span { str: "\"TWO_LEVEL\"", start: 64, end: 75 }, inner: [Pair { rule: string_content, span: Span { str: "TWO_LEVEL", start: 65, end: 74 }, inner: [] }] }, Pair { rule: type_component, span: Span { str: "modifiers = Shift+Lock;", start: 82, end: 105 }, inner: [Pair { rule: modifiers, span: Span { str: "modifiers = Shift+Lock;", start: 82, end: 105 }, inner: [Pair { rule: key_combo, span: Span { str: "Shift+Lock", start: 94, end: 104 }, inner: [Pair { rule: ident, span: Span { str: "Shift", start: 94, end: 99 }, inner: [] }, Pair { rule: ident, span: Span { str: "Lock", start: 100, end: 104 }, inner: [] }] }] }] }, Pair { rule: type_component, span: Span { str: "map[Shift] = Level2;", start: 110, end: 130 }, inner: [Pair { rule: map, span: Span { str: "map[Shift] = Level2;", start: 110, end: 130 }, inner: [Pair { rule: key_combo, span: Span { str: "Shift", start: 114, end: 119 }, inner: [Pair { rule: ident, span: Span { str: "Shift", start: 114, end: 119 }, inner: [] }] }, Pair { rule: ident, span: Span { str: "Level2", start: 123, end: 129 }, inner: [] }] }] }, Pair { rule: type_component, span: Span { str: "preserve[Lock]= Lock;", start: 135, end: 156 }, inner: [Pair { rule: preserve, span: Span { str: "preserve[Lock]= Lock;", start: 135, end: 156 }, inner: [Pair { rule: key_combo, span: Span { str: "Lock", start: 144, end: 148 }, inner: [Pair { rule: ident, span: Span { str: "Lock", start: 144, end: 148 }, inner: [] }] }, Pair { rule: key_combo, span: Span { str: "Lock", start: 151, end: 155 }, inner: [Pair { rule: ident, span: Span { str: "Lock", start: 151, end: 155 }, inner: [] }] }] }] }, Pair { rule: type_component, span: Span { str: "level_name[Level1] = \"Base\";", start: 161, end: 189 }, inner: [Pair { rule: level_name, span: Span { str: "level_name[Level1] = \"Base\";", start: 161, end: 189 }, inner: [Pair { rule: group, span: Span { str: "[Level1]", start: 171, end: 179 }, inner: [Pair { rule: ident, span: Span { str: "Level1", start: 172, end: 178 }, inner: [] }] }, Pair { rule: string, span: Span { str: "\"Base\"", start: 182, end: 188 }, inner: [Pair { rule: string_content, span: Span { str: "Base", start: 183, end: 187 }, inner: [] }] }] }] }, Pair { rule: type_component, span: Span { str: "level_name[Level2] = \"Shift\";", start: 194, end: 223 }, inner: [Pair { rule: level_name, span: Span { str: "level_name[Level2] = \"Shift\";", start: 194, end: 223 }, inner: [Pair { rule: group, span: Span { str: "[Level2]", start: 204, end: 212 }, inner: [Pair { rule: ident, span: Span { str: "Level2", start: 205, end: 211 }, inner: [] }] }, Pair { rule: string, span: Span { str: "\"Shift\"", start: 215, end: 222 }, inner: [Pair { rule: string_content, span: Span { str: "Shift", start: 216, end: 221 }, inner: [] }] }] }] }] }] }]"#).unwrap()
    );
}
