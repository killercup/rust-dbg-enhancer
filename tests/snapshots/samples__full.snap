---
created: "2019-07-01T13:21:59.225753Z"
creator: insta@0.8.1
source: tests/samples.rs
expression: "enhance(r#\"[Pair { rule: xkb_types_item, span: Span { str: \"override type \\\"TWO_LEVEL\\\" {\\n    modifiers = Shift+Lock;\\n    map[Shift] = Level2;\\n    preserve[Lock]= Lock;\\n    level_name[Level1] = \\\"Base\\\";\\n    level_name[Level2] = \\\"Shift\\\";\\n};\", start: 50, end: 226 }, inner: [Pair { rule: type_item, span: Span { str: \"override type \\\"TWO_LEVEL\\\" {\\n    modifiers = Shift+Lock;\\n    map[Shift] = Level2;\\n    preserve[Lock]= Lock;\\n    level_name[Level1] = \\\"Base\\\";\\n    level_name[Level2] = \\\"Shift\\\";\\n};\", start: 50, end: 226 }, inner: [Pair { rule: type_item_prefix, span: Span { str: \"override\", start: 50, end: 58 }, inner: [] }, Pair { rule: string, span: Span { str: \"\\\"TWO_LEVEL\\\"\", start: 64, end: 75 }, inner: [Pair { rule: string_content, span: Span { str: \"TWO_LEVEL\", start: 65, end: 74 }, inner: [] }] }, Pair { rule: type_component, span: Span { str: \"modifiers = Shift+Lock;\", start: 82, end: 105 }, inner: [Pair { rule: modifiers, span: Span { str: \"modifiers = Shift+Lock;\", start: 82, end: 105 }, inner: [Pair { rule: key_combo, span: Span { str: \"Shift+Lock\", start: 94, end: 104 }, inner: [Pair { rule: ident, span: Span { str: \"Shift\", start: 94, end: 99 }, inner: [] }, Pair { rule: ident, span: Span { str: \"Lock\", start: 100, end: 104 }, inner: [] }] }] }] }, Pair { rule: type_component, span: Span { str: \"map[Shift] = Level2;\", start: 110, end: 130 }, inner: [Pair { rule: map, span: Span { str: \"map[Shift] = Level2;\", start: 110, end: 130 }, inner: [Pair { rule: key_combo, span: Span { str: \"Shift\", start: 114, end: 119 }, inner: [Pair { rule: ident, span: Span { str: \"Shift\", start: 114, end: 119 }, inner: [] }] }, Pair { rule: ident, span: Span { str: \"Level2\", start: 123, end: 129 }, inner: [] }] }] }, Pair { rule: type_component, span: Span { str: \"preserve[Lock]= Lock;\", start: 135, end: 156 }, inner: [Pair { rule: preserve, span: Span { str: \"preserve[Lock]= Lock;\", start: 135, end: 156 }, inner: [Pair { rule: key_combo, span: Span { str: \"Lock\", start: 144, end: 148 }, inner: [Pair { rule: ident, span: Span { str: \"Lock\", start: 144, end: 148 }, inner: [] }] }, Pair { rule: key_combo, span: Span { str: \"Lock\", start: 151, end: 155 }, inner: [Pair { rule: ident, span: Span { str: \"Lock\", start: 151, end: 155 }, inner: [] }] }] }] }, Pair { rule: type_component, span: Span { str: \"level_name[Level1] = \\\"Base\\\";\", start: 161, end: 189 }, inner: [Pair { rule: level_name, span: Span { str: \"level_name[Level1] = \\\"Base\\\";\", start: 161, end: 189 }, inner: [Pair { rule: group, span: Span { str: \"[Level1]\", start: 171, end: 179 }, inner: [Pair { rule: ident, span: Span { str: \"Level1\", start: 172, end: 178 }, inner: [] }] }, Pair { rule: string, span: Span { str: \"\\\"Base\\\"\", start: 182, end: 188 }, inner: [Pair { rule: string_content, span: Span { str: \"Base\", start: 183, end: 187 }, inner: [] }] }] }] }, Pair { rule: type_component, span: Span { str: \"level_name[Level2] = \\\"Shift\\\";\", start: 194, end: 223 }, inner: [Pair { rule: level_name, span: Span { str: \"level_name[Level2] = \\\"Shift\\\";\", start: 194, end: 223 }, inner: [Pair { rule: group, span: Span { str: \"[Level2]\", start: 204, end: 212 }, inner: [Pair { rule: ident, span: Span { str: \"Level2\", start: 205, end: 211 }, inner: [] }] }, Pair { rule: string, span: Span { str: \"\\\"Shift\\\"\", start: 215, end: 222 }, inner: [Pair { rule: string_content, span: Span { str: \"Shift\", start: 216, end: 221 }, inner: [] }] }] }] }] }] }]\"#).unwrap()"
---
"[\n    Pair {\n        rule: xkb_types_item,\n        span: Span {\n            str: \"override type \\\"TWO_LEVEL\\\" {\\n    modifiers = Shift+Lock;\\n    map[Shift] = Level2;\\n    preserve[Lock]= Lock;\\n    level_name[Level1] = \\\"Base\\\";\\n    level_name[Level2] = \\\"Shift\\\";\\n};\'\",\n            start: 50,\n            end: 226,\n        },\n        inner: [\n            Pair {\n                rule: type_item,\n                span: Span {\n                    str: \"override type \\\"TWO_LEVEL\\\" {\\n    modifiers = Shift+Lock;\\n    map[Shift] = Level2;\\n    preserve[Lock]= Lock;\\n    level_name[Level1] = \\\"Base\\\";\\n    level_name[Level2] = \\\"Shift\\\";\\n};\'\",\n                    start: 50,\n                    end: 226,\n                },\n                inner: [\n                    Pair {\n                        rule: type_item_prefix,\n                        span: Span {\n                            str: \"override\'\",\n                            start: 50,\n                            end: 58,\n                        },\n                        inner: [],\n                    },\n                    Pair {\n                        rule: string,\n                        span: Span {\n                            str: \"\\\"TWO_LEVEL\\\"\'\",\n                            start: 64,\n                            end: 75,\n                        },\n                        inner: [\n                            Pair {\n                                rule: string_content,\n                                span: Span {\n                                    str: \"TWO_LEVEL\'\",\n                                    start: 65,\n                                    end: 74,\n                                },\n                                inner: [],\n                            },\n                        ],\n                    },\n                    Pair {\n                        rule: type_component,\n                        span: Span {\n                            str: \"modifiers = Shift+Lock;\'\",\n                            start: 82,\n                            end: 105,\n                        },\n                        inner: [\n                            Pair {\n                                rule: modifiers,\n                                span: Span {\n                                    str: \"modifiers = Shift+Lock;\'\",\n                                    start: 82,\n                                    end: 105,\n                                },\n                                inner: [\n                                    Pair {\n                                        rule: key_combo,\n                                        span: Span {\n                                            str: \"Shift+Lock\'\",\n                                            start: 94,\n                                            end: 104,\n                                        },\n                                        inner: [\n                                            Pair {\n                                                rule: ident,\n                                                span: Span {\n                                                    str: \"Shift\'\",\n                                                    start: 94,\n                                                    end: 99,\n                                                },\n                                                inner: [],\n                                            },\n                                            Pair {\n                                                rule: ident,\n                                                span: Span {\n                                                    str: \"Lock\'\",\n                                                    start: 100,\n                                                    end: 104,\n                                                },\n                                                inner: [],\n                                            },\n                                        ],\n                                    },\n                                ],\n                            },\n                        ],\n                    },\n                    Pair {\n                        rule: type_component,\n                        span: Span {\n                            str: \"map[Shift] = Level2;\'\",\n                            start: 110,\n                            end: 130,\n                        },\n                        inner: [\n                            Pair {\n                                rule: map,\n                                span: Span {\n                                    str: \"map[Shift] = Level2;\'\",\n                                    start: 110,\n                                    end: 130,\n                                },\n                                inner: [\n                                    Pair {\n                                        rule: key_combo,\n                                        span: Span {\n                                            str: \"Shift\'\",\n                                            start: 114,\n                                            end: 119,\n                                        },\n                                        inner: [\n                                            Pair {\n                                                rule: ident,\n                                                span: Span {\n                                                    str: \"Shift\'\",\n                                                    start: 114,\n                                                    end: 119,\n                                                },\n                                                inner: [],\n                                            },\n                                        ],\n                                    },\n                                    Pair {\n                                        rule: ident,\n                                        span: Span {\n                                            str: \"Level2\'\",\n                                            start: 123,\n                                            end: 129,\n                                        },\n                                        inner: [],\n                                    },\n                                ],\n                            },\n                        ],\n                    },\n                    Pair {\n                        rule: type_component,\n                        span: Span {\n                            str: \"preserve[Lock]= Lock;\'\",\n                            start: 135,\n                            end: 156,\n                        },\n                        inner: [\n                            Pair {\n                                rule: preserve,\n                                span: Span {\n                                    str: \"preserve[Lock]= Lock;\'\",\n                                    start: 135,\n                                    end: 156,\n                                },\n                                inner: [\n                                    Pair {\n                                        rule: key_combo,\n                                        span: Span {\n                                            str: \"Lock\'\",\n                                            start: 144,\n                                            end: 148,\n                                        },\n                                        inner: [\n                                            Pair {\n                                                rule: ident,\n                                                span: Span {\n                                                    str: \"Lock\'\",\n                                                    start: 144,\n                                                    end: 148,\n                                                },\n                                                inner: [],\n                                            },\n                                        ],\n                                    },\n                                    Pair {\n                                        rule: key_combo,\n                                        span: Span {\n                                            str: \"Lock\'\",\n                                            start: 151,\n                                            end: 155,\n                                        },\n                                        inner: [\n                                            Pair {\n                                                rule: ident,\n                                                span: Span {\n                                                    str: \"Lock\'\",\n                                                    start: 151,\n                                                    end: 155,\n                                                },\n                                                inner: [],\n                                            },\n                                        ],\n                                    },\n                                ],\n                            },\n                        ],\n                    },\n                    Pair {\n                        rule: type_component,\n                        span: Span {\n                            str: \"level_name[Level1] = \\\"Base\\\";\'\",\n                            start: 161,\n                            end: 189,\n                        },\n                        inner: [\n                            Pair {\n                                rule: level_name,\n                                span: Span {\n                                    str: \"level_name[Level1] = \\\"Base\\\";\'\",\n                                    start: 161,\n                                    end: 189,\n                                },\n                                inner: [\n                                    Pair {\n                                        rule: group,\n                                        span: Span {\n                                            str: \"[Level1]\'\",\n                                            start: 171,\n                                            end: 179,\n                                        },\n                                        inner: [\n                                            Pair {\n                                                rule: ident,\n                                                span: Span {\n                                                    str: \"Level1\'\",\n                                                    start: 172,\n                                                    end: 178,\n                                                },\n                                                inner: [],\n                                            },\n                                        ],\n                                    },\n                                    Pair {\n                                        rule: string,\n                                        span: Span {\n                                            str: \"\\\"Base\\\"\'\",\n                                            start: 182,\n                                            end: 188,\n                                        },\n                                        inner: [\n                                            Pair {\n                                                rule: string_content,\n                                                span: Span {\n                                                    str: \"Base\'\",\n                                                    start: 183,\n                                                    end: 187,\n                                                },\n                                                inner: [],\n                                            },\n                                        ],\n                                    },\n                                ],\n                            },\n                        ],\n                    },\n                    Pair {\n                        rule: type_component,\n                        span: Span {\n                            str: \"level_name[Level2] = \\\"Shift\\\";\'\",\n                            start: 194,\n                            end: 223,\n                        },\n                        inner: [\n                            Pair {\n                                rule: level_name,\n                                span: Span {\n                                    str: \"level_name[Level2] = \\\"Shift\\\";\'\",\n                                    start: 194,\n                                    end: 223,\n                                },\n                                inner: [\n                                    Pair {\n                                        rule: group,\n                                        span: Span {\n                                            str: \"[Level2]\'\",\n                                            start: 204,\n                                            end: 212,\n                                        },\n                                        inner: [\n                                            Pair {\n                                                rule: ident,\n                                                span: Span {\n                                                    str: \"Level2\'\",\n                                                    start: 205,\n                                                    end: 211,\n                                                },\n                                                inner: [],\n                                            },\n                                        ],\n                                    },\n                                    Pair {\n                                        rule: string,\n                                        span: Span {\n                                            str: \"\\\"Shift\\\"\'\",\n                                            start: 215,\n                                            end: 222,\n                                        },\n                                        inner: [\n                                            Pair {\n                                                rule: string_content,\n                                                span: Span {\n                                                    str: \"Shift\'\",\n                                                    start: 216,\n                                                    end: 221,\n                                                },\n                                                inner: [],\n                                            },\n                                        ],\n                                    },\n                                ],\n                            },\n                        ],\n                    },\n                ],\n            },\n        ],\n    },\n]"
